### PR TITLE
Exercise grain

### PR DESCRIPTION
**Pull request by pclausen**
Exercise grain added

### CODE REVIEW

File: exercises/practice/grains/grains.f90 [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  implicit none
+
+contains
+
+  ! Using double precision return type instead of int to avoid overflow in int
+  double precision function square(nr)
+    integer :: nr
+    square = -1.d0
+  end function
```
Comment (hiljusti) [inline exercises/practice/grains/grains.f90:12]: 1. I suggest moving the comment out of the stub and into the instructions
2. I don't know that I've ever seen `nr` before, but `n`, `num`, `x`, or `i` are all common variable names for an integer that doesn't need a lot of context in the name

```suggestion
  double precision function square(n)
    integer :: n
    square = -1.d0
  end function
```

File: exercises/practice/grains/grains.f90 [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
```
Comment (hiljusti) [inline exercises/practice/grains/grains.f90:13]: Let's also add a stub for `total`
```suggestion

double precision function total()
  total = -1.d0
end function

```

File: exercises/practice/grains/.docs/instructions.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+king loved chess, the servant told the king he would like to have grains
+of wheat. One grain on the first square of a chess board, with the number
+of grains doubling on each successive square.
+
+There are 64 squares on a chessboard (where square 1 has one grain, square 2 has two grains, and so on).
+
+Write code that shows:
+- how many grains were on a given square, and
+- the total number of grains on the chessboard
+
```
Comment (hiljusti) [inline exercises/practice/grains/.docs/instructions.md:17]: I suggest to put any "you should know before coding" kind of ideas here

```suggestion
Write code that shows:
- how many grains were on a given square, and
- the total number of grains on the chessboard

The function for an individual square will take an integer value (`n`) and both functions will return a double precision value.

```

**Review by hiljusti — CHANGES_REQUESTED** [review]

File: exercises/practice/grains/grains.f90 (hunk 2 above)
Comment (pclausen) [reply exercises/practice/grains/grains.f90:13]: added stub

File: exercises/practice/grains/grains.f90 (hunk 1 above)
Comment (pclausen) [reply exercises/practice/grains/grains.f90:12]: change nr to n

**Review by hiljusti — APPROVED** [review]

**Pull request merged**